Dockerfile

```diff
@@ -0,0 +1,32 @@
+FROM python:3.9-buster as base
+# Perform common operations, dependency installation etc...
+
+ENV POETRY_HOME=$HOME/.poetry
+ENV PATH=$POETRY_HOME/bin:$PATH
+RUN echo "--- Installing Poetry ---"
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+RUN echo "--- Installing dependencies ---"
+
+WORKDIR /todo_app
+COPY pyproject.toml poetry.lock /todo_app/
+
+RUN poetry install
+
+COPY ./todo_app /todo_app/todo_app
+
+FROM base as production 
+# Configure for production
```

Comment:
There are other differences you could make between production and development
E.g. `RUN poetry install --no-dev` for production or `ENV FLASK_ENV=production`


But the ENTRYPOINT was the only required one.